src/model/channel/message.rs

```diff
@@ -264,6 +264,36 @@ impl Message {
         cache_http.http().as_ref().delete_message_reactions(self.channel_id.0, self.id.0).await
     }
 
+    /// Remove reaction from the message given an [`Emoji`] or unicode character
+    /// 
+    /// **Note**: Requires the [Manage Messages] permission, _if_ the current
+    /// user did not perform the reaction
+    /// 
+    /// # Errors
+    /// 
+    /// Returns [`Error::Http`] if the currend user did not perform the reaction,
+    /// and lacks permission
+    /// 
+    /// [Manage Messages]: Permissions::MANAGE_MESSAGES
+    #[inline]
+    pub async fn delete_reaction(
+        self,
+        cache_http: impl CacheHttp,
+        user_id: Option<UserId>,
+        reaction_type: impl Into<ReactionType>,
+    ) -> Result<()> {
+        cache_http
+            .http()
+            .as_ref()
+            .delete_reaction(
+                self.channel_id.0,
+                self.id.0,
+                user_id.map(|uid| uid.0),
+                &reaction_type.into(),
+            )
+            .await
```

Comment:
This should be reworked to just a wrapper around `ChannelId::delete_reaction`. The same goes for `Message::delete_reactions`, now that `ChannelId::delete_reactions` exists.